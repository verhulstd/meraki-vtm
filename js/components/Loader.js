export default class Loader {
  constructor(holder) {
    this._holder = holder;
    this.loaderHTML = this.generateHtml();
  }
  generateHtml() {
    this._holder.insertAdjacentHTML("beforeend", `<p class="loader"></p>`);
    return this._holder.querySelector("p.loader");
  }
  show() {
    this.loaderHTML.style.display = "block";
  }
  hide() {
    this.loaderHTML.style.display = "none";
  }
}
