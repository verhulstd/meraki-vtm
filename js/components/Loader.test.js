import Loader from "./Loader";
document.body.innerHTML = "<div id='holder'></div>";
const holder = document.querySelector("#holder");
const loader = new Loader(holder);

test("Creating a new loader should generate the correct html", () => {
  expect(loader).toMatchSnapshot();
});

test("A hidden loader should have a display of none", () => {
  loader.hide();
  expect(loader.loaderHTML.style.display).toBe("none");
});

test("A shown loader should have a display of block", () => {
  loader.show();
  expect(loader.loaderHTML.style.display).toBe("block");
});
