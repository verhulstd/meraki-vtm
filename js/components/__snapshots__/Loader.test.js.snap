// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creating a new loader should generate the correct html 1`] = `
Loader {
  "_holder": <div
    id="holder"
  >
    <p
      class="loader"
    />
  </div>,
  "loaderHTML": <p
    class="loader"
  />,
}
`;
